### PR TITLE
Fix max retries tests

### PR DESCRIPTION
**Pull request by Rados13**

### CODE REVIEW

File: src/main/kotlin/com/github/davenury/ucac/common/ProtocolTimer.kt [hunk 1]
```diff
@@ -22,17 +23,21 @@ class ProtocolTimerImpl(
...
+                val exponent = 1.5.pow(iteration)
+
                 val backoff = (
                         if (backoffBound.isZero) 0
-                        else randomGenerator.nextLong().absoluteValue % backoffBound.toMillis()
+                        else randomGenerator.nextLong().absoluteValue % (backoffBound.toMillis() * exponent).toLong()
                         )
                     .let { Duration.ofMillis(it) }
                 val timeout = delay.plus(backoff)
+                if (iteration != 0) println("Sleeping for: $timeout /-/ $backoff /-/ $exponent /-/ $iteration")
```
Comment (kjarosh) [inline src/main/kotlin/com/github/davenury/ucac/common/ProtocolTimer.kt:40]: `log.info` + describe the printed parameters

Comment (Davenury) [inline src/main/kotlin/com/github/davenury/ucac/common/ProtocolTimer.kt:40]: logger

File: src/main/kotlin/com/github/davenury/ucac/gpac/domain/GPACProtocol.kt [hunk 2]
```diff
@@ -152,16 +155,34 @@ class GPACProtocolImpl(
...
             }
-        val electResponses = electMePhase(enrichedChange, { responses -> superSet(responses, getPeersFromChange(enrichedChange)) })
+
+        val electMeResult =
+            electMePhase(enrichedChange, { responses -> superSet(responses, getPeersFromChange(enrichedChange)) })
+
+        if (iteration == maxLeaderElectionTries) {
+            logger.error("Transaction failed due to too many retries of becoming a leader.")
+            signal(Signal.ReachedMaxRetries, transaction, getPeersFromChange(change))
+            return
```
Comment (Davenury) [inline src/main/kotlin/com/github/davenury/ucac/gpac/domain/GPACProtocol.kt:175]: shouldn't we throw exception here? Or do we silently fail here?

File: src/main/kotlin/com/github/davenury/ucac/common/ProtocolTimer.kt (hunk 1 above)
Comment (Rados13) [reply src/main/kotlin/com/github/davenury/ucac/common/ProtocolTimer.kt:40]: I will remove it as it was used only for debugging purposes.

Comment (Rados13) [reply src/main/kotlin/com/github/davenury/ucac/common/ProtocolTimer.kt:40]: I will remove it as it was used only for debugging purposes.

File: src/main/kotlin/com/github/davenury/ucac/common/commonRouting.kt [hunk 3]
```diff
@@ -17,10 +18,20 @@ fun Application.commonRouting(
...
+            call.respond(transactionResult)
+        }
+
         post("/create_change") {
             val change = call.receive<Change>()
-            gpacProtocol.performProtocolAsLeader(change)
-            call.respond(HttpStatusCode.OK)
+            val statusCode = when (gpacProtocol.performProtocolAsLeader(change)) {
+                TransactionResult.DONE -> HttpStatusCode.OK
+                TransactionResult.FAILED -> HttpStatusCode.BadRequest
```
Comment (Davenury) [inline src/main/kotlin/com/github/davenury/ucac/common/commonRouting.kt:31]: It's nothing but transaction failed isn't the fault of user (I think), so probably it should be 500 or so

**Review by Davenury — APPROVED** [review]

**Pull request merged**